### ISSUE TITLE
Wrong ordering of serializer and validator in `Annotated` breaks serialized output

### ISSUE THREAD

**Issue by ngnpope** [pydantic/pydantic#10385]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

I've bumped into an issue since updating to v2.9 where a `PlainSerializer` gets called twice when it was only called once under v2.8. The second call results in the wrong value being serialized.

Edit by @Viicos: simplified repro.

```py
from typing import Annotated

from pydantic import BaseModel, PlainSerializer, PlainValidator


def val(v):
    print(f'val {v}')
    return v


def ser(v):
    print(f'ser {v}')
    return v


class Model(BaseModel):
    a: Annotated[int, PlainSerializer(ser), PlainValidator(val)]


m = Model(a=1)
m.model_dump()
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.9.1
        pydantic-core version: 2.23.3
          pydantic-core build: profile=release pgo=false
                 install path: .../pydantic
               python version: 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]
                     platform: Linux-6.9.7-arch1-1-x86_64-with-glibc2.40
             related packages: typing_extensions-4.12.2 pydantic-extra-types-2.9.0 pydantic-settings-2.4.0 mypy-1.1.1
                       commit: ecc5275d
```

**Comment by sydney-runkle** [comment]
@ngnpope,

Thanks for reporting this! We'll work on a fix for v2.9.2!

**Comment by sydney-runkle** [comment]
The bisect is much appreciated 🙏
